src/controllers/admin/mentee.controller.ts

```diff
@@ -0,0 +1,38 @@
+import type { Request, Response } from 'express'
+import { ApplicationStatus, ProfileTypes } from '../../enums'
+import type Profile from '../../entities/profile.entity'
+import type Mentee from '../../entities/mentee.entity'
+import type { ApiResponse } from '../../types'
+import { getAllMentees } from '../../services/admin/mentee.service'
+
+export const getMentees = async (
+  req: Request,
+  res: Response
+): Promise<ApiResponse<Mentee[]>> => {
+  try {
+    const user = req.user as Profile
+    const status: ApplicationStatus | undefined = req.query.status as
+      | ApplicationStatus
+      | undefined
+
+    if (user.type !== ProfileTypes.ADMIN) {
+      return res.status(403).json({ message: 'Only Admins are allowed' })
+    }
+
+    if (status && !(status?.toUpperCase() in ApplicationStatus)) {
```

Comment:
Here we can either remove the optional chaining operator `?`, because you've already checked if it's truthy.
Or it seems using `if (!(status_?.toUpperCase() in ApplicationStatus)) {` also works.